rocDecode-setup.py

```diff
@@ -34,11 +34,11 @@
 __status__ = "Shipping"
 
 # error check calls
-def ERROR_CHECK(call):
-    status = call
-    if(status != 0):
-        print('ERROR_CHECK failed with status:'+str(status))
+def ERROR_CHECK(waitval):
+    if(waitval != 0): # return code and signal flags
```

Comment:
this code is not executing the most important statement which is 
status = call.
Also the following statement is not applicable since it is checking the call/waitval instead of return from the call. The return from the call is int and hence (status != 0) is all that we need to capture.
Not sure what is the purpose of this PR

Reply:
The purpose of the code is to make rocDecode-setup.py exit with a non zero return code if one of the steps wrapped in CHECK_ERROR returns a non zero exit status. 

Take a concrete example. Line 275

            ERROR_CHECK(os.system(
            'sudo zypper ar -cfp 90 \'https://ftp.gwdg.de/pub/linux/misc/packman/suse/openSUSE_Leap_$releasever/Essentials\' packman-essentials'))

This calls the shell to call sudo to call zypper to do something. If zypper fails to do it, it returns a non-zero error code to sudo. sudo in turn passes the error code to the shell. The shell in turn passes the value back to python running this script. This script sees the non-zero code and prints out a message, **but then passes zero back to the program calling this script.** The caller of rocDecode-setup.py has no indication if anything failed that is covered by an ERROR_CHECK. The ERROR_CHECK does print out a message and abort the rest of this python script, but I repeat that the script reports that everything succeeded.

Note that "call" is not a keywork in python. It is just the name of a parameter. In the example line above the string starting with "sudo" and ending with "essentials" is assembled and given to os.system to execute. The return value from os.system, which is an integer, is given to ERROR_CHECK as the value of the call parameter. There is nothing magical about the name ERROR_CHECK, it is a normal python function. I don't understand your comment about this is missing the "status = call". **This is not preprocessor string manipulation**; this is regular python. status is just the name of a local variable unless one of the import statements has declared it as a global, but this would be considered poor programming style to pollute the global namespace.

I changed the name of the parameter from "call" to "waitval" to better reflect the nature the value. If you object to the name change feel free to comment to that effect.